### PR TITLE
Docs: Serve a custom 404 page

### PR DESCRIPTION
**Pull request by chrisjsewell**
Redirect to a helpful page when a document is not found

### CODE REVIEW

**Comment by chrisjsewell** [comment]
this is what it looks like: https://sphinx-notfound-page.readthedocs.io/en/latest/other.html

File: docs/source/404.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+:orphan:
+
+===============
+ Oh, oh -- 404!
+===============
+
+We could not find this page 😢
+
+Please, try using the links from the sidebar to find what you are looking for.
```
Comment (giovannipizzi) [inline docs/source/404.rst:9]: Should we add a comment like 
- If you believe the page you came from has a wrong link, please contact the authors of that page to fix the broken link
- If you believe that the documentation page should be at this link but has been moved by error, you can inform the AiiDA developers by opening an issue on GitHub.

And ideally `an issue on GitHub` should already be a link to an issue with the correct template (i.e. we might want to add a new template, which asks things like:
- broken docs URL
- expected (working) docs URL
- where did you find this link

Comment (sphuber) [reply docs/source/404.rst:9]: And could we find a different icon maybe? I find the excessively crying emoticon a bit jarring tbh

Comment (greschd) [reply docs/source/404.rst:9]: 🤪
¯\_(ツ)_/¯
🤔

Comment (giovannipizzi) [reply docs/source/404.rst:9]: Actually, after discussing with @sphuber we realised that your PR currently only shows a little emoticon and not the huge crying one of the example - so maybe the current suggestion is OK?

File: docs/source/404.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+.. image:: images/logo_aiida_docs.png
+   :align: center
```
Comment (giovannipizzi) [inline docs/source/404.rst:12]: put a max-width?

**Comment by giovannipizzi** [comment]
Another comment by @csadorf - add a search bar and suggest to use the search functionality

File: docs/source/404.rst (hunk 1 above)
Comment (greschd) [reply docs/source/404.rst:9]: Yeah, I guess that's fine.

File: docs/source/404.rst (hunk 2 above)
Comment (chrisjsewell) [reply docs/source/404.rst:16]: done

File: docs/source/404.rst (hunk 1 above)
Comment (chrisjsewell) [reply docs/source/404.rst:9]: done

**Comment by chrisjsewell** [comment]
> add a search bar and suggest to use the search functionality

The search bar will be there, the same as on any other page. Unfortunately you can not view exactly what it looks like until it is actually live on RTD though (because of the way it works)

File: docs/source/404.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+===============
+ Oh, oh -- 404!
+===============
+
+We could not find this page 🤔
+
+Please, try using the links from the sidebar to find what you are looking for.
+
+If you believe that the documentation page should be at this link, or should redirect to a suitable page, we would be grateful if you could inform the AiiDA developers by opening an issue on GitHub:
```
Comment (chrisjsewell) [inline docs/source/404.rst:11]:
```suggestion
If you believe that the documentation page should be at this link, or should redirect to a suitable page, we would be grateful if you could inform the AiiDA developers by opening an issue on GitHub:
```

File: docs/source/404.rst [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ Oh, oh -- 404!
+===============
+
+We could not find this page 🤔
+
+Please, try using the links from the sidebar to find what you are looking for.
+
+If you believe that the documentation page should be at this link, or should redirect to a suitable page, we would be grateful if you could inform the AiiDA developers by opening an issue on GitHub: 
+
+`Click here to open an issue <https://github.com/aiidateam/aiida-core/issues/new?assignees=&labels=topic%2Fdocumentation&template=doc-improvements.md&title=Docs%3A+404>`__
```
Comment (chrisjsewell) [inline docs/source/404.rst:13]:
```suggestion
`Click here to open an issue <https://github.com/aiidateam/aiida-core/issues/new?assignees=&labels=topic%2Fdocumentation&template=doc-improvements.md&title=Docs%3A+404>`__
```

File: docs/source/404.rst (hunk 1 above)
Comment (greschd) [reply docs/source/404.rst:9]: @chrisjsewell I think @giovannipizzi meant the 😢 is fine?

Comment (chrisjsewell) [reply docs/source/404.rst:9]: Oh yeh it was always been the small version, the supersize crying was a bit much lol, but now 🤔

**Review by giovannipizzi — APPROVED** [review]
Ok with me - I'm approving but I'm not sure if there are more things to be checked, also because I guess we cannot test it - if there is an issue we do another PR?

**Comment by chrisjsewell** [comment]
> Ok with me - I'm approving but I'm not sure if there are more things to be checked, also because I guess we cannot test it - if there is an issue we do another PR?

yep, just have to wait and see in the docs. The extension repo is actually part of the RTD organisation, so I'm assuming there should be no issue 🤞

**Review by greschd — APPROVED** [review]

**Comment by greschd** [comment]
> yep, just have to wait and see in the docs

I guess in principle you could have RTD build another branch (in "hidden" mode), but may not be worth the overhead here.

**Comment by chrisjsewell** [comment]
> > yep, just have to wait and see in the docs
> 
> I guess in principle you could have RTD build another branch (in "hidden" mode), but may not be worth the overhead here.

You can actually have RTD run the PR CI directly (I use it in many packages), however its limited to only 2 concurrent builds, so I was tentative to add it to this repo, where we have quite a few PRs at any one time

**Pull request merged**